src/osm_reader/street.rs

```diff
@@ -62,28 +62,34 @@ pub fn streets(
     admins_geofinder: &AdminGeoFinder,
     db_file: &Option<PathBuf>,
     db_buffer_size: usize,
+    settings: &settings::Settings,
 ) -> Result<StreetsVec, Error> {
-    // This is the list of highway that we don't want to index
-    // See [OSM Key Highway](https://wiki.openstreetmap.org/wiki/Key:highway) for background.
-    const INVALID_HIGHWAY: &[&str] =
-        &["bus_guideway", "escape", "bus_stop", "elevator", "platform"];
+    let invalid_highways = &settings
+        .street
+        .clone()
+        .map(|street| street.exclusion.highways.unwrap_or(Vec::new()))
+        .unwrap_or(Vec::new());
```

Comment:
```suggestion
        .map(|street| street.exclusion.highways.unwrap_or_else(Vec::new))
        .unwrap_or_else(Vec::new);
```

That's very minor: Clippy will be likely to suggest avoiding unnecessary function calls here

Reply:
Good call!